subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -2105,6 +2105,32 @@ int lwm2m_engine_delete_res_inst(char *pathstr)
 	return 0;
 }
 
+int lwm2m_engine_path_is_observed(char *pathstr)
+{
+	struct observe_node *obs;
+	struct lwm2m_obj_path path;
+	int ret;
+	int i;
+
+	ret = string_to_path(pathstr, &path, '/');
+	if (ret < 0) {
+		return ret;
+	}
+
+	for (i = 0; i < sock_nfds; ++i) {
+		SYS_SLIST_FOR_EACH_CONTAINER(&sock_ctx[i]->observer, obs, node) {
+			if (obs->path.level <= path.level &&
+			    (obs->path.obj_id == path.obj_id &&
+			     (obs->path.level < 2 ||
```

Comment:
Would be nice to use symbols like `LWM2M_PATH_LEVEL_OBJECT_INST` instead of numbers for clarity.